Tools/ardupilotwaf/boards.py

```diff
@@ -473,8 +473,18 @@ def pre_build(self, bld):
             self.embed_ROMFS_files(bld)
 
     def build(self, bld):
-        bld.ap_version_append_str('GIT_VERSION', bld.git_head_hash(short=True))
-        bld.ap_version_append_int('GIT_VERSION_INT', int("0x" + bld.git_head_hash(short=True), base=16))
+        version_type = 'GIT_VERSION'
+        if version_type in os.environ:
+            bld.ap_version_append_str(version_type, os.environ[version_type])
+        else:
+            bld.ap_version_append_str(version_type, bld.git_head_hash(short=True))
+            
+        version_type_int = 'GIT_VERSION_INT'
+        if version_type_int in os.environ:
+            bld.ap_version_append_str(version_type_int, os.environ[version_type_int])
+        else:
+            bld.ap_version_append_str(version_type_int, int("0x" + bld.git_head_hash(short=True), base=16))
```

Comment:
Should use `ap_version_append_int`

```suggestion
        if version_type_int in os.environ:
            bld.ap_version_append_int(version_type_int, os.environ[version_type_int])
        else:
            bld.ap_version_append_int(version_type_int, int("0x" + bld.git_head_hash(short=True), base=16))
```